core/src/scene/sceneLoader.cpp

```diff
@@ -330,6 +349,236 @@ void SceneLoader::loadTextures(Node textures, Scene& scene) {
+        const MIXES& mixes) {
+
+    std::unordered_set<std::string> uniqueList;
+    Node node;
+    for (const auto& mixStyleName : mixes) {
+        auto mixStyleNode = styles[mixStyleName];
+        if (auto propNode = mixStyleNode[propStr]) {
+            if (auto subPropNode = propNode[subPropStr]) {
+                if (subPropNode.IsMap()) { // for something like shaders: defines or unique
+                    for (const auto& val : subPropNode) {
+                        node[val.first] = val.second;
+                    }
+                } else if (subPropNode.IsScalar()) { // for something like shaders: extension (scalar)
+                    if (uniqueList.find(subPropNode.as<std::string>()) == uniqueList.end()) {
+                        node.push_back(subPropNode);
+                        uniqueList.insert(subPropNode.as<std::string>());
+                    }
+                } else if (subPropNode.IsSequence()) { // for something like shaders: extensions (sequence)
+                    for (const auto& n : subPropNode) {
+                        if (uniqueList.find(n.as<std::string>()) == uniqueList.end()) {
+                            node.push_back(n);
+                            uniqueList.insert(n.as<std::string>());
+                        }
+                    }
+                }
+            }
+        }
+    }
+    return node;
+}
+
+Node SceneLoader::mergeShaderBlocks(Node& styles, const MIXES& mixes) {
+    Node node;
+
+    for (const auto& mixStyleName : mixes) {
+        auto mixStyleNode = styles[mixStyleName];
+        if (auto shadersNode = mixStyleNode["shaders"]) {
+            for (const auto& block : shadersNode["blocks"]) {
+                std::string tag = block.first.as<std::string>();
+                std::string value = block.second.as<std::string>();
+                if (node[tag]) {
+                    node[tag] = node[tag].as<std::string>() + "\n" + value;
+                } else {
+                    node[tag] = value;
+                }
+            }
+        }
+    }
+
+    return node;
+}
+
+Style* SceneLoader::loadStyle(Node& styles, const MIXES& mixes, Scene& scene) {
+
+    Style* style = nullptr;
+
+    Node styleNode;
+
+    // Or`ed style properties
+    for (auto property : {"animated", "texcoords"}) {
+        Node node = propORing(property, styles, mixes);
+        if (!node.IsNull()) { styleNode[property] = node; }
+    }
+
+    // Overwriten style properties (last wins)
+    for (auto property : {"base", "lighting", "texture", "blend"}) {
+        Node node = propOverwrite(property, "", styles, mixes);
+        if (!node.IsNull()) { styleNode[property] = node; }
+    }
+
+
+    // material properties merging
+    Node materialNode;
+    for (auto subProp : {"diffuse", "ambient", "specular", "normal", "shininess"}) {
+        Node node = propOverwrite("material", subProp, styles, mixes);
```

Comment:
Some material properties can themselves be maps, to define texture url, mapping, etc. So overwriting properties won't work you need to merge.

However, it also looks like `propMerge` would not handle merging properties more than one level deep. I would think that ideally `propMerge` would be recursive, so it would properly merge structures of any depth. Another case of that would be uniforms with structures, though we also discussed deprecating/not supporting that in ES. Nonetheless we do have multi-level merging needs for `material` today.